docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,93 @@
+
+##### kube-proxy
+
+By default, with kube-proxy running in `iptables` mode traffic is distributed randomly across all endpoints, regardless of where it originates from. In a 3 zone cluster, traffic is more likely to go to another zone than to stay in the current zone.
+With the `TopologyAwareRouting` feature kube-proxy filters the endpoints it routes to based on the hints in the EndpointSlice resource. In most of the cases, kube-proxy will prefer the endpoint(s) in the same zone. For more details, see the [following section](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#implementation-kube-proxy).
+
+## Topology-aware Services in the Seed cluster
+
+##### etcd-main-client and etcd-events-client
+
+The `etcd-main-client` and `etcd-events-client` Services are topology-aware. They are consumed by the kube-apiserver.
+
+##### kube-apiserver
+
+The `kube-apiserver` Service is topology-aware. It is consumed by the controllers running in the Shoot control plane.
+
+##### gardener-resource-manager
+
+The `gardener-resource-manager` Service that is part of the Shoot control plane is topology-aware. The resource-manager serves webhooks and the Service is consumed by the kube-apiserver for the webhook communication.
+
+##### vpa-webhook
+
+The `vpa-webhook` Service that is part of the Shoot control plane is topology-aware. It is consumed by the kube-apiserver for the webhook communication.
+
+## Topology-aware Services in the Garden cluster
+
+TBD
+
+## How to make a Service topology-aware?
+
+To make a Service topology-aware the following annotation and label has to be added to the Service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.kubernetes.io/topology-aware-hints: "auto"
+  labels:
+    endpoint-slice-hints.resources.gardener.cloud/consider: "true"
+```
+
+The Gardener extensions can use this approach.
+
+## How to enable the `TopologyAwareRouting` feature gate?
+
+The `TopologyAwareRouting` feature gate can be enabled in the component config of the corresponding component. For gardenlet:
+
+```yaml
+apiVersion: gardenlet.config.gardener.cloud/v1alpha1
+kind: GardenletConfiguration
+# ...
+featureGates:
+  TopologyAwareRouting: true
+```
+
+For the garden-operator:
+```yaml
+apiVersion: operator.config.gardener.cloud/v1alpha1
+kind: OperatorConfiguration
+# ...
+featureGates:
+  TopologyAwareRouting: true
+```
+
+> For K8s < 1.24 clusters the `TopologyAwareRouting` feature gate requires the Kubernetes `TopologyAwareHints` feature to be enabled for kube-apiserver, kube-controller-manager and kube-proxy.
```

Comment:
```suggestion
> For K8s < 1.24 clusters, the `TopologyAwareRouting` feature gate requires the Kubernetes `TopologyAwareHints` feature to be enabled for kube-apiserver, kube-controller-manager and kube-proxy.
```

Why? Is it disabled by default? If yes maybe add this here for transparency.

Reply:
This is required because the `TopologyAwareHints` feature gate is disabled by default in K8s < 1.24. When the `TopologyAwareHints` feature gate is disabled the kube-apiserver does not allow anything to be persisted in `.endpoints[].hints` field in the EndpointSlice resource (see https://github.com/kubernetes/kubernetes/blob/3e26e104bdf9d0dc3c4046d6350b93557c67f3f4/pkg/registry/discovery/endpointslice/strategy.go#L124-L157), the kube-controller-manager removes the hints and kube-proxy is not using a topology-aware routing (see https://github.com/kubernetes/kubernetes/blob/b46a3f887ca979b1a5d14fd39cb1af43e7e5d12d/pkg/proxy/topology.go#L140-L186).
I will elaborate this in the doc.